README.md

```diff
@@ -17,95 +18,110 @@ This module adds a new exec provider capable of executing PowerShell commands.
   command   => template('guest/rename-guest.ps1'),
   onlyif    => template('guest/guest-exists.ps1'),
   provider  => powershell,
   logoutput => true,
 }
-```
-Each template is a PowerShell script, 
+~~~
+
+Each template is a PowerShell script.
 
-```
+~~~
 $obj = $(Get-WMIObject Win32_UserAccount -Filter "Name='Guest'")
 $obj.Rename("OtherGuest")
-```
+~~~
 
 This has the added benefit of not requiring escaping '$' in the PowerShell code.
 
 ##Reference
 
-Since the powershell module is a provider for `exec` it enables the same parameters as [`exec`](http://docs.puppetlabs.com/references/stable/type.html#exec) for PowerShell.
-
-###Provider: powershell
+####Provider
+* powershell - Adapts the Puppet `exec` resource to run PowerShell commands.
 
 ####Parameters
+All parameters are optional.
 
-* **command** - The actual PowerShell command to execute. Must either be fully qualified or a search path for the command must be provided. 
-* **creates** - The file to look for before running the command. The command will only run if the file doesn’t exist. *Note:* This parameter will not create a file, it will simply look for one.
-* **cwd** - The directory from which to run the command.
-* **environment** - Additional environment variables to set for a command. Multiple entries should be in an array.
-* **group** - The group to run the command as.  
-* **logoutput** - Whether to log command output in addition to logging the exit code. Valid values are 'true', 'false', and 'on_failure'. Defaults to 'on_failure, which only logs the output when the command has an exit code that does not match any value specified by the returns attribute. 
-* **onlyif** - Only run the exec if the command returns 0.
-* **path** - The search path used for command execution.
-* **refresh** - How to refresh the command.
-* **refreshonly** -  Used with `subscribe` and `notify` [metaparameters](http://docs.puppetlabs.com/references/latest/metaparameter.html) to refresh the command only when a dependent object is changed. Valid values are 'true' and 'false'.
-* **returns** - The expected return code(s). An error will be returned if the executed command returns something else. Defaults to 0. Can be specified as an array of acceptable return codes or a single value.
-* **timeout** - The maximum time the command should take.
-* **tries** - The number of times execution of the command should be tried. Defaults to '1'. 
-* **try_sleep** - The time to sleep in seconds between `tries`.
-* **umask** - The umask to be used while executing the command.
-* **unless** - The `exec` will run unless the command returns 0.
+#####`command`
+Specifies the actual PowerShell command to execute. Must either be fully qualified or a search path for the command must be provided. Valid options: String. Default: Undefined.
 
-##Limitations
+#####`environment`
+Sets additional environment variables to set for a command. Valid options: String, or an array of multiple options. Default: Undefined.
```

Comment:
I don't have a great suggestion yet. Just that string is a type and not a value. But I get that you changed that to valid "options", not values.